.github/workflows/robot/internal/review/review.go

```diff
@@ -0,0 +1,312 @@
+	Owner bool `json:"owner"`
+}
+
+// Config holds code reviewer configuration.
+type Config struct {
+	// Rand is a random number generator. It is not safe for cryptographic
+	// operations.
+	Rand *rand.Rand
+
+	// CodeReviewers and CodeReviewersOmit is a map of code reviews and code
+	// reviewers to omit.
+	CodeReviewers     map[string]Reviewer `json:"codeReviewers"`
+	CodeReviewersOmit map[string]bool     `json:"codeReviewersOmit"`
+
+	// DocsReviewers and DocsReviewersOmit is a map of docs reviews and docs
+	// reviewers to omit.
+	DocsReviewers     map[string]Reviewer `json:"docsReviewers"`
+	DocsReviewersOmit map[string]bool     `json:"docsReviewersOmit"`
+
+	// Admins are assigned reviews when no others match.
+	Admins []string `json:"admins"`
+}
+
+// CheckAndSetDefaults checks and sets defaults.
+func (c *Config) CheckAndSetDefaults() error {
+	if c.Rand == nil {
+		c.Rand = rand.New(rand.NewSource(time.Now().UnixNano()))
+	}
+
+	if c.CodeReviewers == nil {
+		return trace.BadParameter("missing parameter CodeReviewers")
+	}
+	if c.CodeReviewersOmit == nil {
+		return trace.BadParameter("missing parameter CodeReviewersOmit")
+	}
+
+	if c.DocsReviewers == nil {
+		return trace.BadParameter("missing parameter DocsReviewers")
+	}
+	if c.DocsReviewersOmit == nil {
+		return trace.BadParameter("missing parameter DocsReviewersOmit")
+	}
+
+	if c.Admins == nil {
+		return trace.BadParameter("missing parameter Admins")
+	}
+
+	return nil
+}
+
+// Assignments can be used to assign and check code reviewers.
+type Assignments struct {
+	c *Config
+}
+
+// FromString parses JSON formatted configuration and returns assignments.
+func FromString(reviewers string) (*Assignments, error) {
+	var c Config
+	if err := json.Unmarshal([]byte(reviewers), &c); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	r, err := New(&c)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return r, nil
+}
+
+// New returns new code review assignments.
+func New(c *Config) (*Assignments, error) {
+	if err := c.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &Assignments{
+		c: c,
+	}, nil
+}
+
+// IsInternal returns if the author of a PR is internal.
+func (r *Assignments) IsInternal(author string) bool {
+	_, ok := r.c.CodeReviewers[author]
+	return ok
+}
+
+// Get will return a list of code reviewers a given author.
+func (r *Assignments) Get(author string, docs bool, code bool) []string {
+	var reviewers []string
+
+	switch {
+	case docs && code:
+		reviewers = append(reviewers, r.getDocsReviewers(author)...)
+		reviewers = append(reviewers, r.getCodeReviewers(author)...)
+	case !docs && code:
+		reviewers = append(reviewers, r.getCodeReviewers(author)...)
+	case docs && !code:
+		reviewers = append(reviewers, r.getDocsReviewers(author)...)
+	// Strange state, an empty commit? Return admin reviewers.
+	case !docs && !code:
+		reviewers = append(reviewers, r.getCodeReviewers(author)...)
```

Comment:
? 

```suggestion
		reviewers = append(reviewers, r.getAdminReviewers(author)...)
```

Based on comment above

Reply:
+1